src/key/chorded.rs

```diff
@@ -0,0 +1,766 @@
+where
+    K::Context: Into<Context>,
+    K::Event: TryInto<Event>,
+    K::Event: From<Event>,
+{
+    /// Constructs new chorded key.
+    pub const fn new(chord: K, passthrough: K) -> Self {
+        Key { chord, passthrough }
+    }
+
+    /// Constructs new pressed key.
+    pub fn new_pressed_key(
+        &self,
+        context: K::Context,
+        keymap_index: u16,
+    ) -> (
+        input::PressedKey<Self, PressedKeyState<K>>,
+        key::PressedKeyEvents<K::Event>,
+    ) {
+        let mut pk = input::PressedKey {
+            keymap_index,
+            key: *self,
+            pressed_key_state: PressedKeyState::new(context, keymap_index),
+        };
+        let timeout_ev = Event::Timeout;
+        let sch_ev = key::ScheduledEvent::after(
+            context.into().config.timeout,
+            key::Event::key_event(keymap_index, timeout_ev),
+        );
+
+        let mut pke = key::PressedKeyEvents::scheduled_event(sch_ev.into_scheduled_event());
+
+        let n_pke = pk
+            .pressed_key_state
+            .check_resolution(context, keymap_index, self);
+        pke.extend(n_pke);
+
+        (pk, pke)
+    }
+
+    /// Maps the Key of the Key into a new type.
+    pub fn map_key<T: key::Key + Copy>(self, f: fn(K) -> T) -> Key<T> {
+        let Key { chord, passthrough } = self;
+        Key {
+            chord: f(chord),
+            passthrough: f(passthrough),
+        }
+    }
+
+    /// Maps the Key of the Key into a new type.
+    pub fn into_key<T: key::Key + Copy>(self) -> Key<T>
+    where
+        K: Into<T>,
+    {
+        self.map_key(|k| k.into())
+    }
+}
+
+/// Auxiliary chorded key (with a passthrough key).
+///
+/// The auxiliary keys are chorded keys,
+///  but don't store the resolved chord key.
+/// (i.e. After te primary chorded key, the remaining keys
+///  in the chord are defined with auxiliary chorded keys).
+#[derive(Deserialize, Debug, Clone, Copy, PartialEq)]
+pub struct AuxiliaryKey<K> {
+    passthrough: K,
+}
+
+impl<K: key::Key + Copy> AuxiliaryKey<K>
+where
+    K::Context: Into<Context>,
+    K::Event: TryInto<Event>,
+    K::Event: From<Event>,
+{
+    /// Constructs new auxiliary chorded key.
+    pub const fn new(passthrough: K) -> Self {
+        AuxiliaryKey { passthrough }
+    }
+
+    /// Constructs new pressed key.
+    pub fn new_pressed_key(
+        &self,
+        context: K::Context,
+        keymap_index: u16,
+    ) -> (
+        input::PressedKey<Self, PressedKeyState<K>>,
+        key::ScheduledEvent<Event>,
+    ) {
+        let mut pk = input::PressedKey {
+            keymap_index,
+            key: *self,
+            pressed_key_state: PressedKeyState::new(context, keymap_index),
+        };
+        let timeout_ev = Event::Timeout;
+        let sch_ev = key::ScheduledEvent::after(
+            context.into().config.timeout,
+            key::Event::key_event(keymap_index, timeout_ev),
+        );
+        // TODO: return resolved PKE
```

Comment:
Oops, this was left as TODO? Its impl. should be similar to `Key::new_pressed_key`.